search/includes/classes/queue/class-cron.php

```diff
@@ -201,15 +206,47 @@ public function sweep_jobs() {
 		// Check if job has been disabled
 		if ( ! $this->is_enabled() ) {
 			$this->disable_sweeper_job();
+		}
 
+		$this->queue->free_deadlocked_jobs();
+
+		// If job limit has been surpassed, don't schedule any new jobs
+		if ( $this->get_processor_job_count() > self::MAX_PROCESSOR_JOB_COUNT ) {
 			return;
 		}
 
-		$this->queue->free_deadlocked_jobs();
+		$i = 1; // Start incrementor at 1 since the first loop is the first job
+		do {
+			// Prevent loop from scheduling more than the maximum allowed jobs in any single iteration
+			if ( $i > self::MAX_PROCESSOR_JOB_COUNT ) {
+				return;
+			}
+			
+			$this->schedule_batch_job();
 
-		// TODO add a "max batches" setting and keep creating batch jobs until none are found or we hit the max
+			$i++;
+		} while ( $this->get_processor_job_count() < self::MAX_PROCESSOR_JOB_COUNT ); 
+		// Prevent continuing if the current processor job count goes over the max allowed job count
+	}
+
+	public function get_processor_job_count() {
+		// If cron control isn't available, only schedule one job
+		if ( ! class_exists( 'Automattic\\WP\\Cron_Control\\Events_Store' ) ) {
+			return self::MAX_PROCESSOR_JOB_COUNT;
+		}
+
+		global $wpdb;
+
+		$table_name = \Automattic\WP\Cron_Control\Events_Store::instance()->get_table_name();
+
+		$current_processor_job_count = $wpdb->get_var( "SELECT COUNT(*) FROM $table_name WHERE action = 'vip_search_queue_processor' AND status != 'complete'" ); // Cannot prepare table name. @codingStandardsIgnoreLine
+
+		// If null, only schedule one job
+		if ( is_null( $current_processor_job_count ) ) {
+			return self::MAX_PROCESSOR_JOB_COUNT;
```

Comment:
It would be more predictable to return a `0` here, then let the other code handle this situation, since the function is `get_processor_job_count()`. Thoughts?

Reply:
If we're going that route it would make more sense to just return null IMO. 0 implies the cron-control is active and running. 

Getting a null from the processor count means that the table is likely missing meaning cron-control isn't active. Which means we should schedule only one job per round as needed like it did before IMO.

Reply:
Ah ok, I think I understand this a bit better now, and it should be good 👍